app/main.py

```diff
@@ -2,0 +1,69 @@
+class Animal:
+    def __init__(self, name: str, appetite: int, is_hungry: bool = True):
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self):
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self):
+        print(f"Eating {self.appetite} food points...")
+        if self.is_hungry:
+            self.is_hungry = False
+            return self.appetite
+        return 0
+
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry: bool = True):
+        super().__init__(name, appetite=3)
+        self.is_hungry = is_hungry
+
+    def catch_mouse(self):
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry: bool = True):
+        super().__init__(name, appetite=7)
+        self.is_hungry = is_hungry
```

Comment:
Redundant assignment: The 'is_hungry' attribute is already set in the superclass constructor. This line is unnecessary and can be removed.